### ISSUE TITLE
When test successfully passes, private_key disapears

### ISSUE THREAD

**Issue by BerkhanBerkdemir** [ansible/molecule#2478]
## Issue Type

- Bug report

## Molecule and Ansible details

**Ansible**: 2.9.2
**Molecule**: 2.22

Molecule installation method: **pip**  
Ansible installation method: **pip**

## Desired Behavior

When tests successfully passes, I would like to login the instance with 
`molecule login` to check the current status of the configuration.

## Actual Behavior

When tests successfully passes, `molecule login` does not work due to missing 
`private_key`. There is no debug message that states that `private_key` or
similar files are deleted because of successful test result.

If I terminate `molecule` process before finishes the test, the `private_key` file
still resides in its location,
`~/.cache/molecule/ansible-role-lorem/.../private_key`.

[output of `molecule --debug test`](https://gist.github.com/BerkhanBerkdemir/5de1980b55f82584df38a48884e9cbab)

## Additional Context

I use separate role, which is named as `vagrant`, to use Vagrant driver.

## Sample Code

```yaml
---
# molecue/vagrant/molecule.yml
dependency:
  name: galaxy
driver:
  name: vagrant
  provider:
    name: virtualbox
lint:
  name: yamllint
  options:
    config-data:
      ignore: venv/
platforms:
  - name: centos-8
    box: generic/centos8
    interfaces:
      - auto_config: true
        ip: 10.0.0.0
        network_name: private_network
        type: static
provisioner:
  name: ansible
  lint:
    name: ansible-lint
verifier:
  name: testinfra
  lint:
    name: flake8
```

```text
# molecule/vagrant/requirements.txt
molecule[vagrant]==2.22
```

**Comment by ssbarnea** [comment]
Molecule Vagrant driver was moved to https://github.com/pycontribs/molecule-vagrant/issues and if this issue is still valid, please raise a ticket there. Thanks.

**Comment by BerkhanBerkdemir** [comment]
Hi @ssbarnea,

I have looked the link you have provided, and checked the `driver.py`. Maybe I
am missing something, but I could not find anything useful in that repository
that does this delete behavior.

However, I definitely cross-file this unexpected behavior.

**Comment by BerkhanBerkdemir** [comment]
Hi there,

I have just found #2270 which adds a logic that controls `--destroy` flag before
deleting the instances. I wish I would see that in the next 2.x, but that work
was merged to 3.0a5.
